phpstan.neon

```diff
@@ -31,6 +31,9 @@ parameters:
     doctrine:
         objectManagerLoader: tests/object-manager.php
 
+    typeAliases:
+        AwsMetadata: 'array{primary?: mixed, has_instance_store?: mixed, ec2_node?: mixed, region?: mixed}'
```

Comment:
This would be the correct type :) 

```suggestion
        AwsMetadata: 'array{}|array{ec2_node: string, ip: string, region: string, primary: bool, has_instance_store: bool, is_worker: bool, is_web: bool}'
```

Reply:
but why `array{}|array{...}`?

Reply:
Because it's either an empty array or all keys exist, so I think this is preferable to having all keys being optional independently.

Also I'd usually prefer a type def on the class rather than aliases in the phpstan config, keeps things closer together.

Reply:
Yup, started with that. But this one is used at multiple classes. I can easily demote this to class level, If you prefer that way.

Good that we talk about it :D was working on VersionData and VersionDataV2 based on the methods of version entity on global level.

Maybe a separate phpstan-types.yml as include?

Reply:
You can import types from another class, if it's not spread like all over the place IMO this is still clearer than types in an external config.

Reply:
See phpstan-import-type at https://phpstan.org/writing-php-code/phpdoc-types#local-type-aliases

Reply:
True, i'll rebase and rework later 👍